### PR TITLE
Plugin worker + doc + pipe

### PR DESCRIPTION
**Pull request by AnthonySendra**
This PR change several things about plugins:
* Documentation about events
* Add before/after (rename) triggers
* Transform `trigger` into pipe
* Add capability to transform plugin Listener into Worker (if the plugin expose a variable `hooks`)
  * In configuration, just need to add an attribute `threads` and the plugin is now launch in a separated process

### CODE REVIEW

**Comment by scottinet** [comment]
+1

**Comment by j33f** [comment]
+1

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "bunyan": "^1.5.1",
     "cli-color": "^1.1.0",
     "commander": "2.9.0",
+    "easy-circular-list": "^1.0.2",
```
Comment (dbengsch) [inline package.json]: Package version should be bumped to version 1.0.11

**Comment by dbengsch** [comment]
+1

File: docs/plugins.md [hunk 2]
```diff
@@ -93,17 +96,189 @@ The module must have a `package.json` file with a `pluginInfo` entry. The option
...
+|`data:beforeNow`               | `read` | `now` |Triggered before controller `read` and action `now`.|Type: Request object|
+|`data:afterNow`                | `read` | `now` |Triggered after controller `read` and action `now`.|Type: Response object|
+|`data:beforeListIndexes`       | `read` | `listIndexes` |Triggered before controller `read` and action `listIndexes`.|Type: Request object|
+|`data:afterListIndexes`        | `read` | `listIndexes` |Triggered after controller `read` and action `listIndexes`.|Type: Response object|
+|`data:afterServerInfo`         | `read` | `serverInfo` |Triggered after controller `read` and action `serverInfo`.|Type: Response object|
+|`data:beforeCreate`        	| `write` | `create` |Triggered before controller `write` and action `create`.|Type: Request object|
+|`data:afterCreate`         	| `write` | `create` |Triggered after controller `write` and action `create`.|Type: Response object|
+|`data:beforeCreate`        	| `write` | `create` |Triggered before controller `write` and action `create`.|Type: Request object|
```
Comment (dbengsch) [inline docs/plugins.md]: data:beforeCreate and data:afterCreate are documented twice.

Comment (dbengsch) [inline docs/plugins.md]: The table is really badly displayed in github's markdown viewer :(

**Comment by benoitvidis** [comment]
\o/ +1

**Pull request merged**